ccm_web/client/src/components/MultipleUserEnrollmentWorkflow.tsx

```diff
@@ -0,0 +1,354 @@
+        rowNumber: i + 2,
+        email: r[EMAIL_HEADER],
+        role: r[ROLE_HEADER] as ClientEnrollmentType,
+        firstName: r[FIRST_NAME_HEADER],
+        lastName: r[LAST_NAME_HEADER]
+      }))
+
+      setValidEnrollments(externalEnrollments)
+      setActiveStep(CSVWorkflowStep.Review)
+    }
+
+    const handleFile = (file: File): void => {
+      setFile(file)
+      const parser = new FileParserWrapper()
+      parser.parseCSV(
+        file,
+        handleValidation,
+        (message) => setSchemaInvalidations([{ message, type: InvalidationType.Error }])
+      )
+    }
+
+    return (
+      <div>
+        <ExampleFileDownloadHeader
+          description={description}
+          body={requirements}
+          fileName='add_non_um_users.csv'
+          fileData={fileData}
+        />
+        <FileUpload onUploadComplete={handleFile} />
+        <div className={classes.buttonGroup}>
+          <Button variant='outlined' aria-label='Back to select section' onClick={handleBackClick}>Back</Button>
+        </div>
+      </div>
+    )
+  }
+
+  const renderReview = (sectionId: number, enrollments: AddNumberedNewExternalUserEnrollment[]): JSX.Element => {
+    return (
+      <div className={classes.confirmContainer}>
+        {file !== undefined && <CSVFileName file={file} />}
+        <Grid container>
+          <Box clone order={{ xs: 2, sm: 1 }}>
+            <Grid item xs={12} sm={9} className={classes.table}>
+              <BulkEnrollExternalUserConfirmationTable enrollments={enrollments} />
+            </Grid>
+          </Box>
+          <Box clone order={{ xs: 1, sm: 2 }}>
+            <Grid item xs={12} sm={3}>
+              <ConfirmDialog
+                submit={async () => {
+                  await doAddExternalEnrollments(sectionId, enrollments.map(({ rowNumber, ...others }) => others))
+                }}
+                cancel={() => {
+                  handleResetUpload()
+                  setActiveStep(CSVWorkflowStep.Upload)
+                }}
+                disabled={isAddExternalEnrollmentsLoading}
+              />
+            </Grid>
+          </Box>
+        </Grid>
+        <Backdrop className={classes.backdrop} open={isAddExternalEnrollmentsLoading}>
+          <Grid container>
+            <Grid item xs={12}>
+              <CircularProgress color='inherit' />
+            </Grid>
+            <Grid item xs={12}>
+              Loading...
+            </Grid>
+            <Grid item xs={12}>
+              Please stay on the page. This may take up to a couple of minutes for larger files.
+            </Grid>
+          </Grid>
+        </Backdrop>
+      </div>
+    )
+  }
+
+  const renderSuccess = (): JSX.Element => {
+    const settingsLink = (
+      <Link href={props.settingsURL} target='_parent'>Canvas Settings page</Link>
+    )
+    // Need to process actual result here somehow
+    const message = <Typography>New non-UM users have been added to Canvas and your sections!</Typography>
```

Comment:
Update as per @melindakraft `Non-UM Users have been added to the designated sections. New users have been invited to choose a login method.`

Reply:
Should it just be "Non-UM Users have been added to the selected section! New users have been sent an email invitation to choose a login method"? We're only supporting one section right now, I had worded it wrong before.

Reply:
Or really "Non-UM Users have been added to the selected section! New users have also been added to Canvas and sent an email invitation to choose a login method."

Reply:
ok. I will leave it up to you further. my only thing here was friend account is just one aspect of creating Non-UM users. So you can rephrase as you like.

Reply:
Done. See https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/270/commits/be08a25d4f0268b266238d997c5c5324ccec07ac